_rules/descriptive-link-5effbb.md

```diff
@@ -0,0 +1,285 @@
+- [H78: Identifying the purpose of a link using link text combined with its enclosing paragraph](https://www.w3.org/WAI/WCAG21/Techniques/html/H78)
+- [H79: Identifying the purpose of a link in a data table using the link text combined with its enclosing table cell and associated table header cells](https://www.w3.org/WAI/WCAG21/Techniques/html/H79)
+- [H81: Identifying the purpose of a link in a nested list using link text combined with the parent list item under which the list is nested](https://www.w3.org/WAI/WCAG21/Techniques/html/H81)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The link text describes the purpose of the link.
+
+```html
+<a href="#desc">See the description of this product.</a>
+
+<p id="desc">This product consists of several web pages.</p>
+```
+
+#### Passed Example 2
+
+The accessible name describes the purpose of the link.
+
+```html
+<a href="#main" aria-label="Go to the main content of the page">Go</a>
+
+<main>
+	<p id="main">This is the main content.</p>
+</main>
+```
+
+#### Passed Example 3
+
+The link text together with its context describe the purpose of the link.
+
+```html
+<p>To see the description of this product <a href="#desc">click here</a>.</p>
+
+<p id="desc">This product consists of several web pages.</p>
+```
+
+#### Passed Example 4
+
+Both the link text together with its context and the accessible name describe the purpose of the link.
+
+```html
+<p><a href="#" aria-label="Place item in cart">Place item</a><span aria-hidden="true"> in shopping cart.</span></p>
+```
+
+#### Passed Example 5
+
+The accessible name describes the purpose of the link.
+
+```html
+<div role="link" tabindex="0" aria-label="Skip to main content" onclick="document.location+='#main';return false;">
+	Skip
+</div>
+
+<main>
+	<p id="main">This is the main content.</p>
+</main>
+```
+
+#### Passed Example 6
+
+The link text describes the purpose of the link.
+
+```html
+<span role="link" tabindex="0" onclick="document.location+='#desc';return false;">See description of the product.</span>
+
+<p id="desc">This product consists of several web pages.</p>
+```
+
+#### Passed Example 7
+
+The context provided by the list and the link text describe the purpose of the links.
+
+```html
+<ul>
+	<li><a href="https://www.gutenberg.org/files/4300/4300-h/4300-h.htm">Ulysses</a></li>
+	<li>
+		<a href="https://www.gutenberg.org/ebooks/4300.epub.images?session_id=04cd710372888de8d8d322215cdfe8ce5b0f8d73"
+			>EPUB</a
+		>
+	</li>
+	<li><a href="https://www.gutenberg.org/files/4300/4300-0.txt">Plain text</a></li>
+</ul>
+```
+
+#### Passed Example 8
+
+The context provided by the table header and the link text describe the purpose of the links.
+
+```html
+<table>
+	<tr>
+		<th colspan="3">Ulysses</th>
+	</tr>
+	<tr>
+		<td><a href="https://www.gutenberg.org/files/4300/4300-h/4300-h.htm">HTML</a></td>
```

Comment:
Project Gutenberg has been founded in 1971 by the inventor of ebooks. It would be really sad if they went out of business. I believe the urls for the resources are stable. But let me know if you prefer, I can copy the books as test assets since they are free.